Update com-callable-wrapper.md

sharpist opened the pull request:
line 81-93: fixed code syntax
line 138-148: fixed code syntax
line 167-177: fixed code syntax

sharpist committed:
Update com-callable-wrapper.md

line 81-93: fixed code syntax
line 138-148: fixed code syntax
line 167-177: fixed code syntax

richlander commented:
@AaronRobinsonMSFT and I are going to review this PR and document together in the new week. I read most of the doc and found multiple things that could be improved.

This is also a good time to decide what we're going to do w/rt COM Interop docs for the upcoming .NET Core 3.0 release, which will support this feature/scenario.

sharpist commented:
Hello, @richlander and @AaronRobinsonMSFT !
Yes, I want to be part of this. Any of my notes that are useful can be supplemented or corrected and used.
COM Interop documentation may be improved with time including new features in .NET Core 3.0.
Thanks!

svick commented on docs/framework/interop/com-callable-wrapper.md:
```diff
@@ -84,6 +84,8 @@ End Class
 // Implicitly extends System.Object.
 public class Mammal
 {
+    public Mammal() {}
```
If a class doesn't define any other constructor, it will have a default public parameterless constructor. So I don't think adding the constructor here and also below is necessary.

sharpist replied:
You are right, this default public parameterless constructor has been added only as a visual example for perception. It can be deleted.

svick commented on docs/framework/interop/com-callable-wrapper.md:
```diff
@@ -134,9 +136,14 @@ End Class
+{
+    public LoanApp() {}
+
+    Int32 IExplicit.M() { return 0; }
```
I think it's more idiomatic to use `int`, rather than `Int32`. Also, you could use expression-bodied method here:

```suggestion
    int IExplicit.M() => 0;
```

The same applies for the block below.

sharpist replied:
Here I would choose a classic implementation. Because the 'return' instruction is more recognizable and identifiable.

AaronRobinsonMSFT replied:
@sharpist I think the 'classic implementation' argument is valid here, but the type really should be `int`.

sharpist committed:
Update com-callable-wrapper.md

I replaced return type 'Int32' -> 'int'
and 'default public parameterless constructors' now deleted

pkulikov commented on docs/framework/interop/com-callable-wrapper.md:
````diff
@@ -134,9 +134,12 @@ End Class
   
 ```csharp
+// The CLR does not expose a class interface for this type.
+// COM clients can call the members of this class using the methods from the IExplicit interface.
````
This comment is in the C# snippet, so readers with Visual Basic as the selected language will miss that comment. Moreover, comments are not translated, so this content won't be localized. @sharpist please consider moving this paragraph into the body of the article. 

The same is applicable to another snippet update.

sharpist replied:
@pkulikov I think these both accompanying comments can be excluded from the code examples. At your discretion. Because the main article already contains a detailed description.

pkulikov replied:
@sharpist 
>I think these both accompanying comments can be excluded from the code examples.

If the article contains enough explanations, I also don't see the need in the code comments.

pkulikov replied:
@Thraka yes
@sharpist thanks for additional updates

sharpist committed:
Update com-callable-wrapper.md

Removed unnecessary comments from edited code examples.

richlander commented:
@AaronRobinsonMSFT and I were going to meet at 10AM PST on Monday (tomorrow). That looks like the best time (10AM) we can select. DM me on twitter @runfaster2000 and we can work out particulars.

Thraka reviewed (approved):
To me this looks like a good improvement and corrects the problem of not actually implementing the explicit interface in the example
